File: frontend/media-backlog-frontend/src/app/features/media/models/update/UpdateBacklogItemDto.ts
import { CreateMediaDto } from "../create/CreateMediaDto";
import { UpdateMediaDto } from "./UpdateMediaDto";

export interface UpdateBacklogItemDto {
    mediaId: string;
    status: string;
    prioritized: boolean;
    userRating: number;
    recommenders: any[];
    notes: string;
    media: UpdateMediaDto;
}

File: backend/MediaBacklogManagerBackend/MediaBacklogManagerBackend/DTOs/Updating/UpdateBacklogItemDto.cs
using MediaBacklogManagerBackend.Emuns;
using MediaBacklogManagerBackend.Enums;

namespace MediaBacklogManagerBackend.DTOs.Updating
{
    public class UpdateBacklogItemDto
    {
        public int MediaId { get; set; }
        public UserMediaStatus Status { get; set; }
        public bool Prioritized { get; set; }
        public double UserRating { get; set; }
        public string? Notes { get; set; }
        public List<string> Recommenders { get; set; } = new();
        public UpdateMediaDto Media { get; set; }
    }
}
